pkg/services/ngalert/schedule/schedule.go

```diff
@@ -424,7 +424,15 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key ngmodels.AlertR
 	for {
 		select {
 		// used by external services (API) to notify that rule is updated.
-		case <-updateCh:
+		case version := <-updateCh:
```

Comment:
Do rule versions start at 1 or 0?

If they start at 1, what are your thoughts on only skipping the update if the version coming off the channel is `>=1`?

With 0 being the zero-value for int64, this gives it a sense of "only skipping **if the version is provided**" and otherwise behaving as before if the version is set to `ruleVersion`'s default.

I could see a developer who isn't aware of this behavior incorrectly passing the zero-value of `ruleVersion` in to `update`, which would then cause a potentially unexpected failure to update.

Reply:
That is a good idea but I will change this update and remove the suggested "0 behavior" completely. The intention is to get rid of db read and only clean up the state on this `update` message.